build_runner/test/daemon/daemon_test.dart

```diff
@@ -28,6 +27,20 @@ main() {
   final webTarget = DefaultBuildTarget((b) => b..target = 'web');
   final testTarget = DefaultBuildTarget((b) => b..target = 'test');
   var clients = <BuildDaemonClient>[];
+
+  // This needs to be run before setUp since a call to `d.create` will add
```

Comment:
This doesn't reflect my understanding of how the test package works. I would not expect the order of called between this and setup to matter. Instead I think we should be calling addTearDown after creating the client.

Reply:
`d.sandbox` implicitly calls `addTearDown`. Since `package:test` will execute tear downs in order they are added the theory is that this could cause issues as we are mucking with the file system while the daemon is running.

Reply:
We should discuss on Monday. The call to `setUp` is not when this happens, it's when the runner invokes that callback. The order of calls to give the runner these callbacks shouldn't matter, and the comment didn't make sense because the runner will always invoke callbacks from `addTeardown` before the ones from `teardown`

Reply:
Ahh. I understand what you are getting at. Yeah that makes sense and this change wouldn't be required unless we were doing a call to `d.sandbox` directly in the main function. Nonetheless, there is still flakiness I need to track down. I'm convinced that the file lock isn't working as intended on Travis for some reason. I'll add some logging to confirm the theory.